src/WebJobs.Script.Grpc/Extensions/RpcSharedMemoryDataExtensions.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Grpc.Messages;
+using Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.Grpc.Extensions
+{
+    internal static class RpcSharedMemoryDataExtensions
+    {
+        internal static async Task<RpcSharedMemory> ToRpcSharedMemoryAsync(this object value, ILogger logger, string invocationId, ISharedMemoryManager sharedMemoryManager)
+        {
+            if (!sharedMemoryManager.IsSupported(value))
+            {
+                return null;
+            }
+
+            // Put the content into shared memory and get the name of the shared memory map written to
+            SharedMemoryMetadata putResponse = await sharedMemoryManager.PutObjectAsync(value);
+            if (putResponse == null)
+            {
+                logger.LogTrace("Cannot write to shared memory for invocation id: {Id}", invocationId);
+                return null;
+            }
+
+            // If written to shared memory successfully, add this shared memory map to the list of maps for this invocation
+            sharedMemoryManager.AddSharedMemoryMapForInvocation(invocationId, putResponse.Name);
+
+            RpcDataType? dataType = GetRpcDataType(value);
+            if (!dataType.HasValue)
+            {
+                logger.LogTrace("Cannot get shared memory data type for invocation id: {Id}", invocationId);
+                return null;
+            }
+
+            // Generate a response
+            RpcSharedMemory sharedMem = new RpcSharedMemory()
+            {
+                Name = putResponse.Name,
+                Offset = 0,
```

Comment:
is this offset always 0?

Reply:
Currently, yes. The idea is to support reading byte ranges in the future so left a provision for that.